src/bson_mapper/bson_archiver.hpp

```diff
@@ -0,0 +1,1078 @@
+#ifndef CEREAL_ARCHIVES_BSON_HPP_
```

Comment:
Should I just use #pragma once instead? This was from when I copied over Cereal's JSONArchive